tests/wpunit/Fee/PlanListingTest.php

```diff
@@ -0,0 +1,181 @@
+<?php
+/**
+ * Includes tests for the Fees class.
+ */
+
+namespace Fee;
+
+use WPBDP\Tests\WPUnitTestCase;
+use Codeception\Util\Debug;
+use WPBDP__Fee_Plan;
+
+/**
+ * Tests for fee Listing
+ */
+class PlanListingTest extends WPUnitTestCase {
+
+	/**
+	 * @var \WpunitTester
+	 */
+	protected $tester;
+
+	public function testFeePlanVisibility() {
+		$this->tester->wantToTest( 'Test Fee Plan Visibility' );
+		$this->createFees();
+		
+		$this->init_gateway();
+		
+		$this->paymentDisablePlanTest();
+
+		$this->paymentEnabledPlanTest();
+		
+		$this->defaultPlanDisabledPlanTest();
+	}
+
+	/**
+	 * Initialize the gateway.
+	 * Set up Auth net gateway but disable it.
+	 *
+	 */
+	private function init_gateway() {
+		wpbdp_set_option( 'payments-test-mode', true );
+		wpbdp_set_option( 'authorize-net', 0 );
+		$this->assertFalse( 1 === wpbdp_get_option( 'authorize-net' ), 'Gateway Disabled' );
+		// Random details for testing purposes. We won't attempt a charge, we just need to enable payments.
+		wpbdp_set_option( 'authorize-net-login-id', '7h6MbLNyn9qb' );
+		wpbdp_set_option( 'authorize-net-transaction-key', '98GHnS594xy32V7d' );
+	}
+
+	/**
+	 * Test fee plans with disabled gateways.
+	 * This should return a total of 0
+	 */
+	private function paymentDisablePlanTest() {
+		Debug::debug( 'Test disabled payments' );
+		$plans = wpbdp_get_fee_plans();
+		$total = 0;
+		$free_plan = null;
+		foreach ( $plans as $plan ) {
+			$total += absint( $plan->amount );
+			if ( 'free' === $plan->tag ) {
+				$free_plan = $plan;
+			}
+		}
+		$this->assertTrue( ( $total === 0 ), 'Plan total amount is 0' );
```

Comment:
Should we add an expected plan count in both of these tests? And maybe check if the 'third_free' tag exists also?

Reply:
Agreed